crates/ra_hir/src/ty/traits.rs

```diff
@@ -67,10 +67,33 @@ fn solve(
     solution
 }
 
+/// A set of clauses that we assume to be true. E.g. if we are inside this function:
+/// ```rust
+/// fn foo<T: Default>(t: T) {}
+/// ```
+/// we assume that `T: Default`.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Environment {
```

Comment:
Perhaps `TraitEnv` or `WhereEnv`? `Environment` can be confused with name resolution environment, that maps names to declarations